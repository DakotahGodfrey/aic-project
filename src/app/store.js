import { configureStore } from '@reduxjs/toolkit';
import homeReducer from 'app/slices/homeSlice';
import galleryReducer from 'app/slices/gallerySlice';
import artistsReducer from 'app/slices/artistsSlice';
import artworksReducer from 'app/slices/artworkSlice';
import artistReducer from 'app/slices/artistSlice';

export const store = configureStore({
  reducer: {
    home: homeReducer,
    gallery: galleryReducer,
    artists: artistsReducer,
    artwork: artworksReducer,
    artist: artistReducer,
  },
});
