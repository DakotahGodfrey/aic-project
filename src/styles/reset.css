*,
*::before,
*::after {
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
}
html {
  font-size: 10px;
}
ol,
ul {
  list-style: none;
}
a {
  text-decoration: none;
}
table {
  border-collapse: collapse;
  border-spacing: 0;
}
::-webkit-scrollbar {
  width: 0; /* Remove scrollbar space */
  background: transparent; /* Optional: just make scrollbar invisible */
}
