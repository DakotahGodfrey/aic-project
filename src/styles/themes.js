const defaultTheme = {
  textColor: '#222',
  secondaryColor: '#fff',
  navBG: '#111',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#666',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};
const darkTheme = {
  textColor: '#eee',
  navBG: '#111',
  secondaryColor: '#151515',
  lightColor: '#eee',
  brandColor: '#B50938',
  brandColorFaded: 'rgba(181,9,56, 0.7)',
  serifFonts: `Georgia, "Times New Roman", Times, serif`,
  sansSerifFonts: `Helvetica, sans-serif`,
  mutedText: '#ddd',
  linkHover: '#22959c',
  lightGrey: '#F3F3F3',
};

const getFromTheme = (propName) => (props) => props.theme[propName];
export { defaultTheme, darkTheme, getFromTheme };
