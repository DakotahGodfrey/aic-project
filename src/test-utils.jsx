import * as React from 'react';
import { render as rtlRender } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';

function render(ui, { theme, store, ...options } = {}) {
  const Wrapper = ({ children }) => (
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <Router>{children}</Router>
      </ThemeProvider>
    </Provider>
  );
  return rtlRender(ui, { wrapper: Wrapper, ...options });
}

export * from '@testing-library/react';
// override React Testing Library's render with our own
export { render };
