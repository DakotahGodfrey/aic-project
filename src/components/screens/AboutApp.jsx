import React from 'react';
import Feed from 'components/layout/Feed';
import styled from 'styled-components/macro';
import { getFromTheme } from 'styles/themes';
import TextHero from 'components/layout/headers/TextHero';

const AboutContainer = styled.section`
  font-family: ${getFromTheme('sansSerifFonts')};
  color: ${getFromTheme('textColor')};
  h1 {
    font-size: 3.6rem;
    margin-bottom: 2.5rem;
    padding-bottom: 0.5rem;
    border-bottom: 1px solid ${getFromTheme('textColor')};
  }
  h2 {
    font-weight: 200;
    font-size: 3.2rem;
    &:not(:first-child) {
      margin-top: 1.5rem;
    }
  }
  h3 {
    font-size: 2.8rem;
    font-weight: 200;
  }
  p {
    font-size: 2rem;
    padding: 1rem;
    line-height: 1.6;
  }
  a {
    color: ${getFromTheme('brandColor')};
    padding: 0.25rem;
  }
  .socials {
    display: flex;
    flex-direction: column;
    justify-content: center;
    font-size: 1.8rem;
    a {
      display: flex;
      align-items: center;
      i {
        padding: 1.6rem;
      }
    }
  }
`;

const AboutApp = () => {
  document.title = 'AIC Explorer | About this app';
  return (
    <>
      />
      <Feed
        css={`
          min-height: calc(100vh - 13rem);
          max-width: 100rem;
        `}
      >
        <TextHero
          heading={'About'}
          message={
            "AIC Explorer is a tool to explore the Art Institute of Chicago's online archives."
          }
        />
        <AboutContainer>
          <h2>About This App</h2>
          <p>
            I built this app as my final project for the{' '}
            <a href='https://www.codecademy.com/learn/paths/front-end-engineer-career-path'>
              Codecademy Front-End Engineering Pathway.
            </a>{' '}
            I wanted to make something other than the typical To Do List or
            Movies App. I also knew that I wanted the app to have an educational
            theme. I did some reasearch and found the public API offered by The
            Art Institute Of Chicago.
          </p>
          <p>
            While digging through the{' '}
            <a
              rel='noreferrer'
              target='_blank'
              href='https://api.artic.edu/docs/'
            >
              API Docs
            </a>
            , I found several endpoints I could use to provide user's with
            interesting information about the artworks and artists featured in
            the museum all from the comfort of their home. A huge thank you goes
            out to the maintainers of the{' '}
            <a
              rel='noreferrer'
              target='_blank'
              href='https://github.com/art-institute-of-chicago/data-aggregator'
            >
              API
            </a>
            , as well as to the{' '}
            <a rel='noreferrer' target='_blank' href='https://www.artic.edu/'>
              Art Institute Of Chicago
            </a>{' '}
            for offering this amazing resource for free.
          </p>
          <p>
            If you like this project and want to check out some of my other work
            you should visit my portfolio site{' '}
            <a href='https://dakotahg.dev'>dakotahg.dev</a>
          </p>
          <h2>About The Developer</h2>
          <p>
            Hey there! My name is Dakotah Godfrey. I am a community-taught web
            developer from Ontario, Canada. I'm new to the web development
            space, but I’ve become passionate about this field. I've focused on
            building the foundational skills needed to secure a job as a
            frontend developer. Lately, I've built accessible and beautiful web
            apps with React and Redux.
          </p>
          <p>
            Before the Covid-19 epidemic, I was working in the hospitality
            industry in Toronto. As work dried up, I started to look for at a
            career change. I started coding as a hobby, with HTML and CSS.
            Eventually, through resources like{' '}
            <a target='_blank' rel='noreferrer' href='https://codecademy.com'>
              Codecademy
            </a>{' '}
            and a number of online workshops from awesome creators and
            instructors like{' '}
            <a target='_blank' rel='noreferrer' href='https://kentcdodds.com/'>
              {' '}
              Kent C Dodds
            </a>
            ,{' '}
            <a target='_blank' rel='noreferrer' href='https://jhey.dev'>
              Jhey Dev
            </a>
            ,{' '}
            <a
              target='_blank'
              rel='noreferrer'
              href='https://twitter.com/ania_kubow'
            >
              Ania Kubów
            </a>{' '}
            and so many others, I found myself truly enjoying the development
            process. My long-term goal is to help make the tech more accessible.
          </p>
          <p>
            At the moment I'm looking for a position as an intern or junior
            developer, so I can continue to grow my skill set with more
            experience.
          </p>
          <h3>Find me</h3>
          <div className='socials'>
            <div className='social-link'>
              <a href='https://twitter.com/dakotah_dev'>
                <i className='fab fa-twitter fa-2x '></i>
                On twitter @dakotah_dev
              </a>
              <a href='https://github.com/dakotahgodfrey'>
                <i className='fab fa-github fa-2x '></i>
                On GitHub as Dakotah Godfrey
              </a>
              <a href='https://linkedin.com/in/dakotah-godfrey'>
                <i className='fab fa-linkedin-in fa-2x'></i>
                On LinkedIn as Dakotah Godfrey
              </a>
            </div>
          </div>
          <h3>Note</h3>
          <p>
            All images and data used by this app are provided by The Art
            Institue Of Chicago, under a Creative Commons Zero (CC0)
            designation.
          </p>
        </AboutContainer>
      </Feed>
    </>
  );
};

export default AboutApp;
