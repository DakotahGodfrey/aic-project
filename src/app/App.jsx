import React, { useEffect, useState } from 'react';
import { ThemeProvider } from 'styled-components';
import { darkTheme, defaultTheme } from 'styles/themes';
import {
  BrowserRouter as Router,
  Switch,
  Route,
  useLocation,
} from 'react-router-dom';
import SiteNav from 'components/layout/navigation/SiteNav';
import { siteLinks } from 'utils/data';
import Home from 'components/screens/Home';
import Gallery from 'components/screens/Gallery';
import Artists from 'components/screens/Artists';
import Artwork from 'components/screens/Artwork';
import Artist from 'components/screens/Artist';
import AboutApp from 'components/screens/AboutApp';
import NotFound from 'components/screens/NotFound';
import ScrollUpButton from 'components/layout/buttons/ScrollUpButton';

function ScrollToTop() {
  const { pathname } = useLocation();

  useEffect(() => {
    if (window instanceof Window) {
      window.scrollTo(0, 0);
    }
  }, [pathname]);

  return null;
}

const App = () => {
  const [theme, setTheme] = useState('default');

  const handleThemeToggle = () => {
    if (theme === 'default') {
      setTheme('darkTheme');
    } else {
      setTheme('default');
    }
    console.log(theme);
  };
  return (
    <ThemeProvider theme={theme === 'default' ? defaultTheme : darkTheme}>
      <Router>
        <div
          id='app'
          style={
            theme === 'default'
              ? { backgroundColor: '#eee' }
              : { backgroundColor: '#111' }
          }
        >
          <ScrollToTop />
          <SiteNav
            links={siteLinks}
            toggleTheme={handleThemeToggle}
            theme={theme}
          />
          <Switch>
            <Route exact path={'/'} component={Home} />
            <Route exact path={'/gallery'} component={Gallery} />
            <Route exact path={'/artists'} component={Artists} />
            <Route exact path={'/gallery/artwork/:id'} component={Artwork} />
            <Route exact path={'/artists/:id'} component={Artist} />
            <Route exact path={'/about'} component={AboutApp} />
            <Route component={NotFound} />
          </Switch>
          <ScrollUpButton />
        </div>
      </Router>
    </ThemeProvider>
  );
};

export default App;
