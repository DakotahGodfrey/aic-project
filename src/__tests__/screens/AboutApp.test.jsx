import { store } from 'app/store';
import AboutApp from 'components/screens/AboutApp';
import { defaultTheme } from 'styles/themes';
import { render, screen } from 'test-utils';

const heading = 'About';

describe('<AboutApp />', () => {
  it('matches the snapshot', () => {
    const { container } = render(<AboutApp />, {
      theme: defaultTheme,
      store: store,
    });
    expect(container.firstChild).toMatchSnapshot();
  });
  it('renders a heading and message', () => {
    render(<AboutApp />, {
      theme: defaultTheme,
      store: store,
    });
    expect(screen.getAllByRole('heading')[0]).toHaveTextContent(heading);
  });
});
