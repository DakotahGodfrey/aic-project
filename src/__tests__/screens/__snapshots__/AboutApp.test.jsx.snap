// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AboutApp /> matches the snapshot 1`] = `/&gt;`;
